sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidatorTest.java

```diff
@@ -1013,6 +1018,136 @@ void model_with_metadata_label_value_length_has_no_error(int length) {
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @EmptySource
+    void model_has_no_header_names_specified_has_error(String headerName) {
+        /* prepare */
+        List<HTTPHeaderConfiguration> httpHeaders = createListOfHeaders(headerName, "secret-key", "https://mywebapp.com");
+
+        SecHubWebScanConfiguration webScan = new SecHubWebScanConfiguration();
+        webScan.headers = Optional.ofNullable(httpHeaders);
+        webScan.url = URI.create("https://mywebapp.com");
+
+        SecHubConfigurationModel model = new SecHubConfigurationModel();
+        model.setApiVersion("1.0");
+        model.setWebScan(webScan);
+
+        modelSupportCollectedScanTypes.add(ScanType.WEB_SCAN);
+
+        /* execute */
+        SecHubConfigurationModelValidationResult result = validatorToTest.validate(model);
+
+        /* test */
+        assertHasError(result, SecHubConfigurationModelValidationError.WEB_SCAN_NO_HEADER_NAME_DEFINED);
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @EmptySource
+    void model_has_no_header_values_specified_has_error(String headerValue) {
+        /* prepare */
+        List<HTTPHeaderConfiguration> httpHeaders = createListOfHeaders("Authorization", headerValue, "https://mywebapp.com");
+
+        SecHubWebScanConfiguration webScan = new SecHubWebScanConfiguration();
+        webScan.headers = Optional.ofNullable(httpHeaders);
+        webScan.url = URI.create("https://mywebapp.com");
+
+        SecHubConfigurationModel model = new SecHubConfigurationModel();
+        model.setApiVersion("1.0");
+        model.setWebScan(webScan);
+
+        modelSupportCollectedScanTypes.add(ScanType.WEB_SCAN);
+
+        /* execute */
+        SecHubConfigurationModelValidationResult result = validatorToTest.validate(model);
+
+        /* test */
+        assertHasError(result, SecHubConfigurationModelValidationError.WEB_SCAN_NO_HEADER_VALUE_DEFINED);
+    }
+
+    private SecHubWebScanConfiguration createWebScanConfigurationWithHeader(String targetUrl, String onlyForUrl) {
+        String headerName = "Authorization";
+        String headerValue = "secret-key";
+
+        List<HTTPHeaderConfiguration> httpHeaders = createListOfHeaders(headerName, headerValue, onlyForUrl);
+
+        SecHubWebScanConfiguration webScan = new SecHubWebScanConfiguration();
+        webScan.headers = Optional.ofNullable(httpHeaders);
+        webScan.url = URI.create(targetUrl);
+
+        return webScan;
+    }
+
+    private List<HTTPHeaderConfiguration> createListOfHeaders(String headerName, String headerValue, String onlyForUrl) {
+        HTTPHeaderConfiguration httpHeader = new HTTPHeaderConfiguration();
+        httpHeader.setName(headerName);
+        httpHeader.setValue(headerValue);
+        httpHeader.setOnlyForUrls(Optional.ofNullable(Arrays.asList(onlyForUrl)));
+        List<HTTPHeaderConfiguration> httpHeaders = new ArrayList<>();
+        httpHeaders.add(httpHeader);
+
+        return httpHeaders;
+    }
+
```

Comment:
Could you please ad an additional tests here with multiple header configuration entries ?

There should be also one with multiple headers and multiple onlyForURLs (to test "matrix" works as well)